power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -328,13 +329,18 @@ inline auto get_edge_weights(TransformerGraph const& graph) -> TrafoGraphEdgePro
         // side via the bidirectional edge (if it exists). For delta configuration ABC, the above
         // situations can happen.
         // The logic still holds in meshed grids, albeit operating a more complex graph.
-        if (edge_src_rank != edge_tgt_rank - 1) {
-            throw AutomaticTapInputError("The control side of a transformer regulator should be relatively further "
-                                         "away from the source than the tap side.\n");
+        if (is_unreachable(edge_res)) {
+            continue;
         }
-        if (!is_unreachable(edge_res)) {
-            result.emplace_back(graph[e].regulated_idx, edge_tgt_rank);
+        if (edge_src_rank == infty && edge_tgt_rank != infty) {
+            throw AutomaticTapInputError(
+                "The transformer is being controlled from non source side towards source side which is unsupported.\n");
         }
+        if (edge_src_rank != edge_tgt_rank - 1) {
+            edge_res = last_rank;
+        }
```

Comment:
probably good to explain this in a comment, e.g.
```suggestion
        if (edge_src_rank != edge_tgt_rank - 1) {
            // Control side is also controlled by a closer regulated transformer.
            // Make this transformer have the lowest possible priority.
            edge_res = last_rank;
        }
```